lib/ansible/module_utils/network/common/facts/facts.py

```diff
@@ -0,0 +1,130 @@
+
+        self.ansible_facts = {'ansible_network_resources': {}}
+        self.ansible_facts['ansible_gather_network_resources'] = list()
+        self.ansible_facts['ansible_net_gather_subset'] = list()
+
+        if not self._gather_subset:
+            self._gather_subset = ['!config']
+        if not self._gather_network_resources:
+            self._gather_network_resources = ['all']
+
+    def gen_runable(self, subsets, valid_subsets):
+        """ Generate the runable subset
+
+        :param module: The module instance
+        :param subsets: The provided subsets
+        :param valid_subsets: The valid subsets
+        :rtype: list
+        :returns: The runable subsets
+        """
+        runable_subsets = set()
+        exclude_subsets = set()
+        minimal_gather_subset = frozenset(['default'])
+
+        for subset in subsets:
+            if subset == 'all':
+                runable_subsets.update(valid_subsets)
+                continue
+            if subset == 'min' and minimal_gather_subset:
+                runable_subsets.update(minimal_gather_subset)
+                continue
+            if subset.startswith('!'):
+                subset = subset[1:]
+                if subset == 'min':
+                    exclude_subsets.update(minimal_gather_subset)
+                    continue
+                if subset == 'all':
+                    exclude_subsets.update(
+                        valid_subsets - minimal_gather_subset)
+                    continue
+                exclude = True
+            else:
+                exclude = False
+
+            if subset not in valid_subsets:
+                self._module.fail_json(msg='Bad subset')
+
+            if exclude:
+                exclude_subsets.add(subset)
+            else:
+                runable_subsets.add(subset)
+
+        if not runable_subsets:
+            runable_subsets.update(valid_subsets)
+        runable_subsets.difference_update(exclude_subsets)
+
+        return runable_subsets
+
+    def get_network_resources_facts(self, netres_choices, facts_resource_obj_map, resource_facts_type=None, data=None):
+        """
+        :param netres_choices:
+        :param fact_resource_subsets:
+        :param data: previously collected configuration
+        :return:
+        """
+        if netres_choices:
+            if 'all' in netres_choices:
+                netres_choices.remove('all')
+
+        if netres_choices:
+            if not resource_facts_type:
+                resource_facts_type = self._gather_network_resources
+
+            restorun_subsets = self.gen_runable(resource_facts_type, netres_choices)
+            if restorun_subsets:
+                self.ansible_facts['gather_network_resources'] = list(restorun_subsets)
+                instances = list()
+                for key in restorun_subsets:
+                    fact_cls_obj = facts_resource_obj_map.get(key)
+                    if fact_cls_obj:
```

Comment:
```suggestion
                    if facts_resource_obj_map.get(key):
```

Reply:
the value of `facts_resource_obj_map.get(key)` is stored in `fact_cls_obj` hence it is not required to get(key) again in if statement.

Reply:
Ah, by respective suggestion I meant was of using the `facts_resource_obj_map.get(key)` directly, without storing into a variable and then comparing.

Reply:
`fact_cls_obj` is used later on at line 103, hence IMO it's better to store it instead of referencing it twice.